### DISCUSSION TITLE
Create stubs for pypdf

### DISCUSSION THREAD

**Discussion by frague59** [py-pdf/pypdf#1716] [Q&A]
## Explanation

I'm using pypdf for an internal project, mypy complains about stubs not present for this package

## Code Example

```python
import pypdf
```

```sh
$ mypy ./src
src/<...>.py:18: error: Cannot find implementation or library stub for module named "pypdf"  [import]

```

**Comment by MartinThoma** [comment]
I just converted this to a discussion, as I'm rather certain this is not a pypdf issue. I guess it's something about your local setup. Very likely, you have a old version of pypdf in your environment (used by mypy) or maybe a file called `pypdf.py`?

Please try the following:

```
mkdir pypdf-mypy
cd pypdf-mypy
python -m venv venv
source venv/bin/activate

python -m pip install pypdf --upgrade
python -m pip install mypy --upgrade
python --version
```

Then create a file called "example.py" and copy the following in there:

```
from pypdf import PdfReader

reader = PdfReader("example.pdf")
page = reader.pages[0]
print(page.extract_text())
```

Now run `mypy .` in that folder. Do you still see that error?

If yes, we can re-open the issue. (but with WAY more information).

If no, you have something wrong on your system.

**Comment by frague59** [comment]
Thanks for your attention ! 

I've tests your example, and it works... I'll explore my setup.

**Reply by MartinThoma** [reply]
There might also be a `.pyc` file or a file in a `__pycache__` directory messing with you. Good luck :-)
